components/builder-originsrv/src/data_store.rs

```diff
@@ -1813,6 +1825,83 @@ impl DataStore {
         oin.set_names(names);
         oin
     }
+    pub fn create_origin_secret(
+        &self,
+        os: &originsrv::OriginSecretCreate,
+    ) -> SrvResult<originsrv::OriginSecret> {
+        let conn = self.pool.get(os)?;
+        let rows = conn.query(
+            "SELECT * FROM insert_origin_secret_v1($1, $2, $3)",
+            &[
+                &(os.get_secret().get_origin_id() as i64),
+                &os.get_secret().get_name(),
+                &os.get_secret().get_value(),
+            ],
+        ).map_err(SrvError::OriginSecretCreate)?;
+        match rows.iter().nth(0) {
+            Some(row) => Ok(self.row_to_origin_secret(row)),
+            None => Err(SrvError::NoRowsReturnedAfterInsert()),
+        }
+    }
+
+    fn row_to_origin_secret(&self, row: postgres::rows::Row) -> originsrv::OriginSecret {
+        let mut os = originsrv::OriginSecret::new();
+        let os_id: i64 = row.get("id");
+        os.set_id(os_id as u64);
+        let os_origin_id: i64 = row.get("origin_id");
+        os.set_origin_id(os_origin_id as u64);
+        os.set_name(row.get("name"));
+        os.set_value(row.get("value"));
+        os
+    }
+
+    pub fn get_origin_secret(
+        &self,
+        os_get: &originsrv::OriginSecretGet,
+    ) -> SrvResult<Option<originsrv::OriginSecret>> {
+        let conn = self.pool.get(os_get)?;
+        let rows = &conn.query(
+            "SELECT * FROM get_origin_secret_v1($1, $2)",
+            &[&(os_get.get_origin_id() as i64), &os_get.get_name()],
+        ).map_err(SrvError::OriginSecretGet)?;
+        if rows.len() != 0 {
+            // We just checked - we know there is a value here
```

Comment:
All the comments justifying `unwrap()`s are one of my favorite parts of our code.  😆